ts/indicators/slow-stochastic.src.ts

```diff
@@ -0,0 +1,191 @@
+     * and `stock/indicators/stochastic.js` files.
+     *
+     * @sample stock/indicators/slow-stochastic
+     *         Slow Stochastic oscillator
+     *
+     * @extends      plotOptions.stochastic
+     * @since        8.0.0
+     * @product      highstock
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/stochastic
+     * @requires     stock/indicators/slowstochastic
+     * @optionparent plotOptions.slowstochastic
+     */
+    {
+        params: {
+            /**
+             * Periods for Slow Stochastic oscillator: [%K, %D, SMA(%D)].
+             *
+             * @type    {Array<number,number,number>}
+             * @default [14, 3, 3]
+             */
+            periods: [14, 3, 3]
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        nameBase: 'Slow Stochastic',
+        init: function (this: Highcharts.SlowStochasticIndicator): void {
+            const args = arguments,
+                ctx = this;
+
+            requiredIndicator.isParentLoaded(
+                (H.seriesTypes.stochastic as any),
+                'stochastic',
+                ctx.type,
+                function (indicator: Highcharts.Indicator): undefined {
+                    indicator.prototype.init.apply(ctx, args);
+                    return;
+                }
+            );
+        },
+        getValues: function<TLinkedSeries extends Highcharts.Series> (
+            this: Highcharts.SlowStochasticIndicator,
+            series: TLinkedSeries,
+            params: Highcharts.SlowStochasticIndicatorParamsOptions
+        ): (Highcharts.IndicatorValuesObject<TLinkedSeries>|undefined) {
+            const periods: Array<number> = (params.periods as any),
+                fastValues = seriesTypes.stochastic.prototype.getValues.call(
+                    this,
+                    series,
+                    params
+                ) as any,
+                slowValues = {
+                    values: [] as Array<Array<number>>,
+                    xData: [] as Array<number>,
+                    yData: [] as Array<Array<number>>
+                };
+
+            let i = 0;
+
+            if (!fastValues) {
+                return;
+            }
+
+            slowValues.xData = fastValues.xData.slice(periods[1] - 1);
+            const fastYData = fastValues.yData.slice(periods[1] - 1);
+
+            // Get SMA(%D)
+            const smoothedValues: undefined|
+            Highcharts.IndicatorValuesObject<Highcharts.Series> =
+                seriesTypes.sma.prototype.getValues.call(
+                    this,
+                    ({
+                        xData: slowValues.xData,
+                        yData: fastYData
+                    } as any),
+                    {
+                        index: 1,
+                        period: periods[2]
+                    }
+                );
```

Comment:
Can I do something with this little monster? Especially lines 137-139.. @bre1470

Reply:
Please see `ts/README.md` and make use of parentheses with type unions. We have to discuss the increase of line characters though.

Reply:
Ahh, I see, will do, thanks!